src/internal_modules/roc_sndio/pump.cpp

```diff
@@ -74,7 +74,7 @@ status::StatusCode Pump::run() {
 
     roc_log(LogDebug, "pump: exiting main loop");
 
-    return code;
+    return sink_.flush();
```

Comment:
If code was not StatusOK before flushing, we should return original code. Only if there were no previous errors before flush and code was StatusOK, we should return code from flush.

(It is a good practice to return earliest error happened, because later errors can be caused by the very first one).

Currently, we just drop previous code, which is wrong.

Reply:
Also:

* please move flush call before "exiting main loop" message
* if flush fails, please log error and status (see status::code_to_str)

Reply:
Done